cstrml/tee/src/lib.rs

```diff
@@ -257,42 +255,18 @@ impl<T: Trait> Module<T> {
     /// Get updated workload by controller account,
     /// this function should only be called in the new era
     /// otherwise, it will be an void in this recursive loop
-    fn update_and_get_workload(controller: &T::AccountId, current_rs: u64) -> u128 {
-        // 1. Get current era
-        let last_rs = current_rs - REPORT_SLOT;
-
-        // 2. Judge if this controller reported works in the former era
-        if let Some(wr) = Self::work_reports((controller, last_rs)) {
-            // a. Remove former era's work report
-            <WorkReports<T>>::remove((controller, last_rs));
-
-            // b. Did report works in the last era
-            // ...
-            // 889(600): (123,300){wr300} ✅ | (123, 300){wr0} ❌
-            // 889(600): (123, 300){wr300} -> (123, 600){wr300}
-            // 1080(900): (123, 600){wr600} ✅ | (123, 600){wr300} ❌
-            // 1080(900): (123, 600){wr600} -> (123, 900){wr600}
-            // ...
-            // old(old_old) <- new
-            // new(old) <- new_new
-            // TODO: between the extended current_rs wr and the real current_rs wr contains a void,
-            // it may mislead storage order
-            if wr.block_number == last_rs {
-                // Extend the last work report(if not exist) into this new report slot
-                if !<WorkReports<T>>::contains_key((controller, current_rs)) {
-                    // This should already be true
-                    <WorkReports<T>>::insert((controller, current_rs), &wr);
-                }
-                (wr.empty_workload + wr.meaningful_workload) as u128
-            // c. Did not report anything in the last era
+    fn update_and_get_workload(controller: &T::AccountId, current_rs: u64) -> (u128, u128) {
+        // 1. Judge if this controller reported works in this current era
+        if let Some(wr) = Self::work_reports(controller) {
+            if Self::reported_in_slot(controller, current_rs) {
+                (wr.empty_workload as u128, wr.meaningful_workload as u128)
             } else {
-                // Cut workload
-                MeaningfulWorkload::mutate(|mw| *mw -= wr.meaningful_workload as u128);
-                EmptyWorkload::mutate(|ew| *ew -= wr.empty_workload as u128);
-                0
+                // Remove work report
+                <WorkReports<T>>::remove(controller);
```

Comment:
300-600 no wr.
650. new wr
660 end of era. new wr would be deleted.
920 end of new era. Cannot find wr.
950 new wr.
1220 end of new era. wr is found and valid. work come back. 

I recommend checking the wr.block_number to see whether it's a new work report or not before deleting it.

Reply:
that make sense, this will make new reporter confused.